doc/website/blog/2021-08-27-visualizing-your-data-graph-using-entviz.md

```diff
@@ -0,0 +1,162 @@
+
+### Introducing entviz
+[entviz](https://github.com/hedwigz/entviz) is an ent extension that automatically generates a static HTML page that visualizes your data graph.
+![Entviz example output](https://link/to/image)
+Most ER diagram generation tools need to connect to your database and introspect it, which makes it harder to maintain an up-to-date diagram of the DB. Since entviz integrates directly to your Ent schema, it does not need to connect to your database, and it automatically generates fresh visualization every time you modify your schema.
+
+If you want to know more about ent how entviz was implemented, checkout the [implementation section](#implementation).
+
+  
+### Usage
+First, let’s add entviz extension to our entc.go file:
+```bash
+go get github.com/hedwigz/entviz
+```
+:::info
+If you are not familiar with `entc` you're welcome to read [entc documentation](https://entgo.io/docs/code-gen#use-entc-as-a-package) to learn more about it.
+:::
+```go title="ent/entc.go"
+err := entc.Generate("./schema", &gen.Config{}, entc.Extensions(entviz.Extension{}))
+ if err != nil {
+   log.Fatalf("running ent codegen: %v", err)
+ }
+```
+Let’s say we have a simple schema with a user entity, and some fields:
+```go title="ent/schema/user.go"
+// Fields of the User.
+func (User) Fields() []ent.Field {
+ return []ent.Field{
+   field.String("name"),
+   field.String("email"),
+   field.Time("created").
+     Default(time.Now),
+ }
+}
+```
+Now, entviz will automatically generate a visualization of our graph everytime we run: 
+```bash
+go generate ./...
+```
+You should now see a new file called `schema-viz.html` in your ent directory
+```bash
+ll ./ent/schema-viz.html
+-rw-r--r-- 1 hedwigz hedwigz 7.3K Aug 27 09:00 schema-viz.html
+```
+Open the html file with your favorite browser to see the visualization
+![tutorial image](https://link/to/image)
+Next, let’s add another entity - Post, and see how our visualization
+```bash
+ent init Post
+```
+```go title="ent/schema/post.go"
+// Fields of the Post.
+func (Post) Fields() []ent.Field {
+ return []ent.Field{
+   field.String("content"),
+   field.Time("created").
+     Default(time.Now),
+ }
+}
+```
+Now we add an edge ([O2M](http://localhost:3000/docs/schema-edges/#o2m-two-types))  from User to Post:
+```go title="ent/schema/post.go"
+// Edges of the User.
+func (User) Edges() []ent.Edge {
+ return []ent.Edge{
+   edge.To("posts", Post.Type),
+ }
+}
+```
+Finally, regenerate the code:
+```bash
+go generate ./…
+```
+Refresh your browser to see the updated result! 
+![tutorial image 2](https://link/to/image)
+
+### Implementation
+Entviz was implemented by extending ent via its extension API.
+Ent extension API lets you aggregate multiple [templates](https://entgo.io/docs/templates/), [hooks](https://entgo.io/docs/hooks/), [options](https://entgo.io/docs/code-gen/#code-generation-options) and [annotations](https://entgo.io/docs/templates/#annotations).
```

Comment:
I am unsure whether this should be part of the post about `entviz` (pretty neat extension btw). I, however, have little experince with log posts, just wanted to leave this here :-)